### PR TITLE
Add removal script for snapshot removal

### PR DESCRIPTION
**Pull request by reyntjensw**
I have added a cleanup script in order to remove old snapshots after a fixed amount of time

### CODE REVIEW

File: snapshot-replicator/functions/remove_snapshots.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import boto3  
+import datetime  
+import argparse
+
+TARGET_REGION = 'eu-central-1'  
+DURATION= 30
+iam = boto3.client('iam')  
+instances = ['frisket-production-api-rds01']
```
Comment (SamClinckspoor) [inline snapshot-replicator/functions/remove_snapshots.py]: this shouldn't be hardcoded.

File: snapshot-replicator/functions/remove_snapshots.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import boto3  
+import datetime  
+import argparse
+
+TARGET_REGION = 'eu-central-1'  
+DURATION= 30
```
Comment (SamClinckspoor) [inline snapshot-replicator/functions/remove_snapshots.py]: can you make this configurable?

File: snapshot-replicator/functions/remove_snapshots.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+def lambda_handler(event, context):
+    target = boto3.client('rds', region_name=TARGET_REGION)
+    for instance in instances:
+        for snapshot in target.describe_db_snapshots(DBInstanceIdentifier=instance, MaxRecords=50)['DBSnapshots']:
+            create_ts = snapshot['SnapshotCreateTime'].replace(tzinfo=None)
+            if create_ts < datetime.datetime.now() - datetime.timedelta(days=DURATION):
+                print "Deleting snapshot id:", snapshot['DBSnapshotIdentifier']
+                target.delete_db_snapshot(
+                    DBSnapshotIdentifier=snapshot['DBSnapshotIdentifier']
+                )
```
Comment (SamClinckspoor) [inline snapshot-replicator/functions/remove_snapshots.py]: Can you add a newline at the end here

**Review by SamClinckspoor — CHANGES_REQUESTED** [review]

File: snapshot-replicator/functions/remove_snapshots.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import boto3  
+import datetime  
+import argparse
```
Comment (SamClinckspoor) [inline snapshot-replicator/functions/remove_snapshots.py]: you don't seem to be using this pakage

File: snapshot-replicator/functions/create_snapshot.py [hunk 5]
```diff
@@ -1,14 +1,11 @@
...
 import re
 import os
-import datetime
 
-#SOURCE_REGION = os.environ['SOURCE_REGION']
-SOURCE_REGION = "eu-west-1"  
-iam = boto3.client('iam')  
-instances = ["frisket-production-api-rds01"]
+SOURCE_REGION = os.environ['SOURCE_REGION']
+instances = os.environ['DB_INSTANCES']
```
Comment (SamClinckspoor) [inline snapshot-replicator/functions/create_snapshot.py]: can you update your code to either use all caps for env variables or all lowercase? atm its mixed with eg `SOURCE_REGION` and `instances`

**Review by SamClinckspoor — APPROVED** [review]
small nitpick

File: snapshot-replicator/functions/shipper.py [hunk 6]
```diff
@@ -1,13 +1,13 @@
-import boto3  
-import botocore  
+import boto3
```
Comment (SamClinckspoor) [inline snapshot-replicator/functions/shipper.py]: Can you fix indentation in this file? you are mixing 2 and 4 spaces

File: snapshot-replicator/functions/shipper.py [hunk 7]
```diff
@@ -42,7 +42,7 @@ def lambda_handler(event, context):
                 SourceDBSnapshotIdentifier=source_snap_arn,
                 TargetDBSnapshotIdentifier=target_snap_id,
                 SourceRegion=SOURCE_REGION,
-                KmsKeyId='arn:aws:kms:eu-central-1:353795632189:key/14f47fc7-a9f1-4889-90fd-e10660bb0c9b',
+                KmsKeyId=KMS_KEY_ID,
                 CopyTags = True)
                 print(response)
             except botocore.exceptions.ClientError as e:
```
Comment (SamClinckspoor) [inline snapshot-replicator/functions/shipper.py]: you removed the botocore package

**Review by venturel — APPROVED** [review]

**Pull request merged**